BotRoyaleUnity/Assets/Scripts/PartHandler.cs

```diff
@@ -107,4 +119,44 @@ public int greatestDirectionStrength()
         }
         return index;
     }
+
+    public void SetCenterPos(int x, int y)
+    {
+        centerPos = new Vector2Int(x, y);
+    }
+
+    public Vector2Int GetCenterPos()
+    {
+        return centerPos;
+    }
+
+    public void EmitCurrentParts()
+    {
+        JSONObject data = new JSONObject();
+        data["action"] = "currentParts";
+        data["name"] = gameObject.name;
+        JSONArray parts = new JSONArray();
+        foreach (Transform part in gameObject.transform)
+        {
+            var partHealth = part.gameObject.GetComponent<PartHealth>();
+            JSONObject newPart = new JSONObject();
+            newPart["type"] = partHealth.GetPartType();
+            newPart["x"] = (partHealth.GetRelPos().x + centerPos.x);
+            newPart["y"] = (partHealth.GetRelPos().y + centerPos.y);
+            newPart["direction"] = partHealth.GetPartDirection();
+            newPart["health"] = partHealth.GetHealth();
+            parts.Add(newPart);
+        }
+        data["parts"] = parts;
+        socket.Emit("game-message", data.ToString());
```

Comment:
Try using the method `EmitGameMessage` in SocketConnectionHandler so that the socket library doenst need to called directly